src/hotspot/share/gc/shared/collectedHeap.hpp

```diff
@@ -551,4 +556,30 @@ class GCCauseSetter : StackObj {
   }
 };
 
+// Class to compute the total CPU time for a set of threads, then update an
+// hsperfdata counter.
+
+class ThreadTotalCPUTimeClosure: public ThreadClosure {
```

Comment:
This class is not really related to GC (and you are already using it to get the CPU time of the `VMThread`) so I wonder if we should put this in a more generic location (maybe `thread.hpp` or something similar). This will become even more relevant if we implement CPU time for JIT compiler threads because I don't think we want to have a dependency on `collectedHeap.hpp` there.